src/main/java/com/example/todolist/Domain/Validation/UserValidation/EmailRule.java

```diff
@@ -0,0 +1,25 @@
+package com.example.todolist.Domain.Validation.UserValidation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import org.hibernate.validator.constraints.UniqueElements;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+@NotNull
+@NotBlank
+@Email(message = "Email must be written in format example@email.com")
```

Comment:
note this, we will deep dive into it, 
NOTE: the message, it will not always "Email must be ..."